File: examples/o3r/configuration/configuration.py
import json

# Define the ifm3d objects for the communication
from ifm3dpy.device import O3R
o3r = O3R()

# Get the current configuration
config = o3r.get()

# Print a little part from the config to verify the configuration
print(json.dumps(config['device']['swVersion'], indent=4))
# Note: this assumes that a camera is plugged into port 1
print(config['ports']['port1']['state'])

# Let's change the name of the device
o3r.set({'device':{'info':{'name':'great_o3r'}}})
o3r.save_init()

# Double check the configuration
config = o3r.get()
print(config['device']['info']['name'])

File: examples/o3r/configuration/configuration.cpp
/*
 * Copyright 2021-present ifm electronic, gmbh
 * SPDX-License-Identifier: Apache-2.0
 */

#include <iostream>
#include <iomanip>
#include <memory>
#include <fstream>
#include <ifm3d/device/o3r.h>
using json = ifm3d::json;


int main(){

    // Create the camera object
    auto cam = std::make_shared<ifm3d::O3R>();

    // Get the current configuration of the camera in JSON format
    json conf = cam->Get();

    // Display the current configuration
    std::cout << std::setw(4) << conf << std::endl;
    
    // Configure the device from a json string
    cam->Set(ifm3d::json::parse(R"({"device":{"info":{"name": "my_new_o3r"}}})")); 
    // Make the configuration persistent
    cam->SaveInit();

    return 0;
}